compiler/rustc_trait_selection/src/traits/dyn_compatibility.rs

```diff
@@ -639,10 +639,10 @@ fn object_ty_for_trait<'tcx>(
 /// a pointer.
 ///
 /// In practice, we cannot use `dyn Trait` explicitly in the obligation because it would result in
-/// a new check that `Trait` is dyn-compatible, creating a cycle (until dyn_compatible_for_dispatch
-/// is stabilized, see tracking issue <https://github.com/rust-lang/rust/issues/43561>).
```

Comment:
For the record, I don't think that this is even true. Even if `dyn_compatible_for_dispatch` was stabilized, if we used `dyn Trait` rather than a placeholder here, we'd still have a query cycle if the `DispatchFromDyn` goal we register below needs to prove that `dyn Trait: Trait`.